pkg/source/kafka/kafka_source.go

```diff
@@ -0,0 +1,361 @@
+//
+// Copyright (c) 2020-present Snowplow Analytics Ltd. All rights reserved.
+//
+// This program is licensed to you under the Snowplow Community License Version 1.0,
+// and you may not use this file except in compliance with the Snowplow Community License Version 1.0.
+// You may obtain a copy of the Snowplow Community License Version 1.0 at https://docs.snowplow.io/community-license-1.0
+
+package kafkasource
+
+import (
+	"context"
+	"crypto/sha256"
+	"crypto/sha512"
+	"fmt"
+	"hash"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/Shopify/sarama"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"github.com/twinj/uuid"
+	"github.com/xdg/scram"
+
+	"github.com/snowplow/snowbridge/config"
+	"github.com/snowplow/snowbridge/pkg/common"
+	"github.com/snowplow/snowbridge/pkg/models"
+	"github.com/snowplow/snowbridge/pkg/source/sourceiface"
+)
+
+// Configuration configures the source for records
+type Configuration struct {
+	Brokers        string `hcl:"brokers" env:"SOURCE_KAFKA_BROKERS"`
+	TopicName      string `hcl:"topic_name" env:"SOURCE_KAFKA_TOPIC_NAME"`
+	ConsumerName   string `hcl:"consumer_name" env:"SOURCE_KAFKA_CONSUMER_NAME"`
+	OffsetsInitial int64  `hcl:"offsets_initial" env:"SOURCE_KAFKA_OFFSETS_INITIAL"`
+
+	ConcurrentWrites int    `hcl:"concurrent_writes,optional" env:"SOURCE_CONCURRENT_WRITES"`
+	Assignor         string `hcl:"assignor,optional" env:"SOURCE_KAFKA_ASSIGNOR"`
+	TargetVersion    string `hcl:"target_version,optional" env:"SOURCE_KAFKA_SOURCE_VERSION"`
+	EnableSASL       bool   `hcl:"enable_sasl,optional" env:"SOURCE_KAFKA_ENABLE_SASL"`
+	SASLUsername     string `hcl:"sasl_username,optional" env:"SOURCE_KAFKA_SASL_USERNAME" `
+	SASLPassword     string `hcl:"sasl_password,optional" env:"SOURCE_KAFKA_SASL_PASSWORD"`
+	SASLAlgorithm    string `hcl:"sasl_algorithm,optional" env:"SOURCE_KAFKA_SASL_ALGORITHM"`
+	TLSCert          string `hcl:"tls_cert,optional" env:"SOURCE_KAFKA_TLS_CERT_B64"`
+	TLSKey           string `hcl:"tls_key,optional" env:"SOURCE_KAFKA_TLS_KEY_B64"`
+	TLSCa            string `hcl:"tls_ca,optional" env:"SOURCE_KAFKA_TLS_CA_B64"`
```

Comment:
nit: We should name these the same way we do for the target: https://github.com/snowplow/snowbridge/blob/623842cbf785ea30ab71fa974b4cf58cd42e299a/pkg/target/kafka.go#L42-L44

(they expect filepaths rather than a previous iteration which was b64)